src/main/java/io/github/thebusybiscuit/slimefun4/implementation/guide/BookSlimefunGuide.java

```diff
@@ -215,7 +217,11 @@ private void addSlimefunItem(Category category, int page, Player p, PlayerProfil
 
             ChatComponent component = new ChatComponent(ChatUtils.crop(ChatColor.RED, item.getItemName()) + "\n");
             component.setHoverEvent(new HoverEvent(ChatColor.RESET + item.getItemName(), ChatColor.DARK_RED.toString() + ChatColor.BOLD + SlimefunPlugin.getLocalization().getMessage(p, "guide.locked"), "", ChatColor.GREEN + "> Click to unlock", "", ChatColor.GRAY + "Cost: " + ChatColor.AQUA.toString() + research.getCost() + " Level(s)"));
-            component.setClickEvent(new ClickEvent(key, player -> research(player, profile, item, research, category, page)));
+            component.setClickEvent(new ClickEvent(key, player ->
+                    SlimefunPlugin.runSync(() -> {
+                        research.guideClickInteraction(this, player, profile, item, category, page);
+                    })
```

Comment:
Having brackets on a one-liner would be redundant.
Also why don't you just do runSync() inside your interaction method?

```suggestion
                    SlimefunPlugin.runSync(() -> 
                        research.guideClickInteraction(this, player, profile, item, category, page);
                    )
```